prism/src/main/java/org/apache/falcon/resource/channel/HTTPChannel.java

```diff
@@ -170,8 +171,20 @@ private String pathValue(Method method, Object... args) throws FalconException {
                     String annotationClass = paramAnnotation.annotationType().getName();
 
                     if (annotationClass.equals(QueryParam.class.getName())) {
-                        queryString.append(getAnnotationValue(paramAnnotation, "value")).
-                                append('=').append(arg).append("&");
+                        if (args[index] instanceof List){
+                            List lifecycle = (List) args[index];
```

Comment:
I believe we cannot handle it generically here because the nature of the list that we can get in jax rs can be very different and might not implement a to string method. We might end up getting a run time exception.

Also just to point out we can refactor it right now code is not generic enough we have checks like args[index] instanceof String || args[index] instanceof Boolean || args[index] instanceof Integer
                    || args[index] instanceof List 
tomorrow a set might also come.

I am open for suggestions if you have a better way to implement it.